### DISCUSSION TITLE
a lot of json source generator errors after updating to 8.0.200

### DISCUSSION THREAD

**Discussion by ivanjx** [dotnet/runtime#99107] [General]
so i have been working on a wpf project since dotnet 8.0.100 preview 7. today i just upgraded my sdk into dotnet 8.0.200 and now my json source generators are complaining with `CS0102` and `CS0111` with 0 code changes. my system text json package version is on 8..0.2. i tried cleaning the solution, deleting bin and obj, rebooting the pc, but none works. any idea on how to fix it? the only way to make the project run again now is to remove every single one of the json source generators. interestingly i also have mvvm toolkit source generator and regex source generator with 0 issues.

**Comment by huoyaoyuan** [comment]
> i tried cleaning the solution, deleting bin and obj, rebooting the pc, but none works.

Source generator output is determined by your code and the version of generator. New SDK brings new generator and the output will deterministically be new version.

> interestingly i also have mvvm toolkit source generator and regex source generator with 0 issues.

The generators are running fine. It's just outputting code that doesn't compile fine. The output of different generators are independent.

Can you paste some of your data models that triggers bad generated code?

@eiriktsarpalis

**Comment by ivanjx** [comment]
@huoyaoyuan even this below cause it to fail. i even removed everything else except for this one below and yet still error.
```
[JsonSerializable(typeof(Dictionary<string, string>))]
public partial class TheContext : JsonSerializerContext
{
}
```

**Reply by ivanjx** [reply]
note that if i create a new project and use json generator it works fine. just specific to my wpf project i was working on.

**Reply by huoyaoyuan** [reply]
`CS0102` and `CS0111` are about duplicate items with same names. You can figure out what name is duplicated from error message, then use symbolic search (Ctrl+T in Visual Studio) to find which files are declaring such names.

**Reply by ivanjx** [reply]
@huoyaoyuan i made sure that that single context is the only one in my project. when i removed `TheContext` it can build again. this project is small so i am sure im not skipping over anything.

**Reply by ivanjx** [reply]
also the class `TheContext` is not reused anywhere else in the code

**Reply by huoyaoyuan** [reply]
What's the exact message of error CS0102/CS0111? The class isn't applicable to the error code. Only its member is.

**Reply by ivanjx** [reply]
```
1>myproject\System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\TheContext.DictionaryStringString.g.cs(13,142,13,165): error CS0102: The type 'TheContext' already contains a definition for '_DictionaryStringString'
1>myproject\System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\TheContext.DictionaryStringString.g.cs(18,140,18,162): error CS0102: The type 'TheContext' already contains a definition for 'DictionaryStringString'
1>myproject\System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\TheContext.DictionaryStringString.g.cs(23,141,23,170): error CS0111: Type 'TheContext' already defines a member called 'Create_DictionaryStringString' with the same parameter types
1>myproject\System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\TheContext.DictionaryStringString.g.cs(43,22,43,60): error CS0111: Type 'TheContext' already defines a member called 'DictionaryStringStringSerializeHandler' with the same parameter types
```

**Reply by ivanjx** [reply]
it seems like there are two or more `TheContext` source is being created

**Reply by huoyaoyuan** [reply]
Yes. Can you do a symbolic search for `DictionaryStringString`? VS should find generated source.

**Reply by ivanjx** [reply]
@huoyaoyuan sorry but how to do symbolic search in vs?

**Reply by huoyaoyuan** [reply]
Ctrl+T. Symbols in generated files should also appear.

**Reply by ivanjx** [reply]
i tried to find one of the property's references and this is all i got:
![image](https://github.com/dotnet/runtime/assets/12814796/6726ba87-ed3b-499f-ac5c-fbcc2bc49cd5)

**Reply by ivanjx** [reply]
also there does not seem to be any duplicates according to vs:
![image](https://github.com/dotnet/runtime/assets/12814796/34efcdf3-4e8b-43f5-95f4-0b7dd00c05d5)

**Reply by ivanjx** [reply]
what do these numbers mean? `TheContext.DictionaryStringString.g.cs(13,142,13,165)`

**Reply by colejohnson66** [reply]
(starting line, starting character, ending line, ending character). So, `(13,142,13,165)` means line 13 from characters 142 to 165.

**Reply by eiriktsarpalis** [reply]
Sharing a minimal reproduction might help here. Start by slowly removing code from the project in a way that doesn't impact the error. Once you get to a point where no more changes can be made without making the error going away, you have a minimal reproduction.

**Reply by ivanjx** [reply]
well thats another problem for me because creating a new project and adding the stj source generator works fine. even other projects in the same solution works fine. this is very localized to the specific wpf project i have been working on since dotnet 8 preview 7

**Reply by elgonzo** [reply]
Make a **full** backup of the solution directory containing the project. Then, and only then, start removing stuff from the project you are working on as @eiriktsarpalis suggested. After finishing troubleshooting (and hopefully having discovered the cause of the problem you observe) restore the solution directory to its current state from the backup you made. If the error could not be successfully be diagnosed or resolved, before restoring the solution directory make a copy of it, which is then the minimal reproduction that can be shared (make sure that it doesn't contain private or other things you don't want to share...)
